x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -321,6 +323,40 @@ protected SpatialRelatesFunction rule(SpatialRelatesFunction function) {
         }
     }
 
+    static class ReplaceOrderByExpressionWithEval extends OptimizerRules.OptimizerRule<OrderBy> {
+
+        @Override
+        protected LogicalPlan rule(OrderBy orderBy) {
+            int counter = 0;
```

Comment:
I wonder if this counter maybe should be a (static?) class member instead of a local var?

I could see multiple subsequent sorts accidentally shadowing each other's temporary evals in some edge cases/weird interactions with other opt. rules - by making the name+counter essentially unique, we'd be avoiding this.